File: Stack/Q3. Exclusive Time of Functions/exclusiveTime.js
/**
 * @param {number} n
 * @param {string[]} logs
 * @return {number[]}
 */
var exclusiveTime = function (n, logs) {
    let stk = []
    let ans = Array(n).fill(0)
    let lastTime = 0

    for (let oper of logs) {
        let [id, type, pos] = oper.split(":")
        // let id = item[0]
        // let type = item[1]
        // let pos = Number(item[2])

        if (type === "start") {
            if (stk.length) {
                ans[stk[stk.length - 1]] += Number(pos) - lastTime
            }
            lastTime = Number(pos)
            stk.push(id)
        }
        else {
            let last = stk.pop()
            ans[last] += Number(pos) - lastTime + 1
            lastTime = Number(pos) + 1
        }

    }
    return ans
};

File: Stack/Q3. Exclusive Time of Functions/exclusiveTime.cpp
#include <iostream>
#include <vector>
#include <string>
#include <stringstream>

using namespace std;

class Solution {
public:
    vector<int> exclusiveTime(int n, vector<string>& logs) {
        vector<string> stk;
        vector<int> ans(n,0);
        int prevTime = 0;

        for(auto& item : logs){
            string id, type, time;
            stringstream ss(item);

            getline(ss, id, ':');
            getline(ss, type, ':');
            getline(ss, time, ':');

            if(type == "start"){
                if(stk.size()){
                    cout<< stk.back();
                    ans[stoi(stk.back())] += stoi(time) - prevTime;
                }
                stk.push_back(id);
                prevTime = stoi(time);
            }
            else{
                ans[stoi(stk.back())] += stoi(time) - prevTime + 1;
                stk.pop_back();
                prevTime = stoi(time) + 1;
            }
        }

        return ans;
    }
};